csrc/id_model/validation_utils.cpp

```diff
@@ -0,0 +1,144 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <compute_at_map.h>
+#include <id_model/to_string.h>
+#include <id_model/validation_utils.h>
+
+#include <sstream>
+
+namespace nvfuser {
+
+// Compare the given exact graph with ComputeAtMap. Their maps should
```

Comment:
Nice comments! For public methods, I tend to put comments in `.h` not `.cpp` for easier discovery.

Reply:
Agreed, that's what I usually do. Not sure again why I did this way, haha.